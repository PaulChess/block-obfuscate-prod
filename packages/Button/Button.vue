<template>
 <div class="hxmui-button hxmui-fontsize-adapter"
  :style="[getConatainerWidth, customContainerStyle]">
   <button
    :class="[styleClasses, customClass]"
    @click="handleClick"
    @touchstart="()=>{}"
    :disabled="disabledStatus">
    <div :class="getBemClass('content')"
      @click="handleClick">
      <!-- normal -->
      <span class="text" v-if="!loading">
        <slot></slot>
      </span>
      <!-- loading -->
      <div class="loading-block" v-if="loading">
        <hxm-loading :color="loadingColor" :size="loadingSize"></hxm-loading>
        <span class="loading-text" v-if="loadingText">{{loadingText}}</span>
      </div>
      <!-- icon -->
      <div class="icon-block" v-if="icon && !loading">
        <img class="icon" :src="icon" :style="{width: iconWidth}">
        <span class="icon-text" v-if="iconText">{{iconText}}</span>
      </div>
    </div>
  </button>
  </div>
</template>

<script lang="ts">
import {
  Component,
  Prop,
  Vue,
} from 'vue-property-decorator';
import SetFontSize from '../mixins/SetFontSize';
import hxmLoading from '../Loading';
import { createNamespace } from '../utils/create/index';

const [bem] = createNamespace('button');

@Component({
  mixins: [SetFontSize],
  name: 'hxmButton',
  components: {
    hxmLoading,
  },
})

export default class HxmButton extends Vue {
  @Prop({ default: 'primary' }) private type!: string;

  @Prop({ default: 'normal' }) private size!: string;

  @Prop({ default: false }) private disabled!: boolean;

  @Prop({ default: false }) private round!: boolean; // 圆形

  @Prop({ default: false }) private square!: boolean; // 方形

  @Prop({ default: false }) private loading!: boolean;

  @Prop({ default: '' }) private loadingText!: string;

  @Prop({ default: '0.16em' }) private loadingSize!: string;

  @Prop({ default: '#fff' }) private loadingColor!: string;

  @Prop({ default: '' }) private icon!: string;

  @Prop({ default: '0.16em' }) private iconWidth!: string;

  @Prop({ default: '' }) private iconText!: string;

  @Prop({ default: '' }) private customClass !: string;

  @Prop() private customContainerStyle !: any;

  get styleClasses() {
    return bem([
      this.type,
      this.size,
      {
        round: this.round,
        square: this.square,
        disabled: this.disabled,
        loading: this.loading,
      },
    ]);
  }

  get getConatainerWidth() {
    let widthStyle = {};
    if (this.size === 'large') {
      widthStyle = { width: '100%' };
    } else if (this.size === 'medium') {
      widthStyle = { width: '50%' };
    }
    return widthStyle;
  }

  get disabledStatus() {
    let disabled = false;
    if (this.loading || this.disabled) {
      disabled = true;
    }
    return disabled;
  }

  getBemClass(params: any) {
    return bem(params);
  }

  handleClick() {
    this.$emit('click');
  }
}
</script>
