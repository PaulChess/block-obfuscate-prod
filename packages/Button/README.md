# Button 按钮 <Badge text="1.1.3+" />

## 简介

<card>

### 介绍
按钮用于触发一个操作，如点击跳转

</card>

## 代码演示

<card>

### 按钮类型

按钮支持`primary`、`light`、`plain`、`grey`、`link`五种类型，默认为`primary`。
::: warning 注意
正常情况下按钮支持`light`类型，但由于light类型目前没有适配`disabled`样式，示例暂不展示
:::

```html
<hxm-button type="primary">主要按钮</hxm-button>
<hxm-button type="plain">朴素按钮</hxm-button>
<hxm-button type="grey">灰色按钮</hxm-button>
<hxm-button type="link">文字按钮</hxm-button>
```
</card>

<card>

### 禁用状态

通过`disabled`属性来禁用按钮，禁用状态下按钮不可点击，但依然会暴露`click`事件供开发者使用

```html
<hxm-button type="primary" disabled="true">主要按钮</hxm-button>
<hxm-button type="plain" disabled="true">朴素按钮</hxm-button>
<hxm-button type="grey" disabled="true">灰色按钮</hxm-button>
<hxm-button type="link" disabled="true">文字按钮</hxm-button>
```
</card>

<card>

### 按钮尺寸

支持`large`、`medium`、`subMedium`、`normal`、`small`、`mini`六种尺寸, 默认为`normal`

```html
<hxm-button size="large">大号按钮</hxm-button>
<hxm-button size="medium">中号按钮</hxm-button>
<hxm-button size="subMedium">次中按钮</hxm-button>
<hxm-button size="normal">常规按钮</hxm-button>
<hxm-button size="small">小号按钮</hxm-button>
<hxm-button size="mini">迷你按钮</hxm-button>
```
</card>

<card>

### 按钮形状

通过`square`设置方形按钮，通过`round`设置圆形按钮

```html
<hxm-button square="true">方形按钮</hxm-button>
<hxm-button round="true">圆形按钮</hxm-button>
```
</card>

<card>

### 加载状态
通过`loading`属性将按钮设置为加载状态，加载状态下默认会隐藏按钮文字，可以通过`loading-text`设置加载状态下的文字

::: tip 提示
`1.5.0`版本以后支持将icon或普通文本与loading状态合写进一个标签
:::

```html
<hxm-button loading="true"></hxm-button>
<hxm-button loading="true" loading-text="加载中..."></hxm-button>
```
</card>

<card>

### 图标按钮

通过`icon`属性设置按钮图标,当前仅支持传入URL; 通过`icon-text`设置图标按钮文案

```html
<hxm-button
  type="plain"
  icon="https://testm.10jqka.com.cn/cd/mbweb-vaserviece-strategy-api-container/newjuece_v3/img/top1.89c5c19.png">
</hxm-button>
<hxm-button
  type="plain"
  icon-text="榜1"
  icon="https://testm.10jqka.com.cn/cd/mbweb-vaserviece-strategy-api-container/newjuece_v3/img/top1.89c5c19.png">
</hxm-button>
```
</card>

<card>

### 点击事件

按钮组件暴露了`click`事件供开发者自定义事件

```html
 <hxm-button @click="handleClick">点击按钮</hxm-button>
```
</card>

<card>

### 自定义样式

如果出现按钮组件内置样式不支持的情况，如`颜色`,`padding`,`圆角`等，用户可以通过`customClass`属性传入自定义类名覆盖按钮样式

```html
<hxm-button customClass="attention">按钮</hxm-button>
<hxm-button customClass="has-attentioned">按钮</hxm-button>
```
```css
// less
.attention {
  background-color: rgba(233, 48, 48, 0.1);
  color: #e93030;
  [theme-mode="black"] & {
    background-color: rgba(233, 48, 48, 0.1);
    color: #fd4332;
  }
}
.has-attentioned {
  background-color: #f6f6f6;
  color: #999;
  [theme-mode="black"] & {
    background-color: #252525;
    color: #8e8e8e;
  }
}
```
</card>

## API

<card>

### props

| 参数 | 说明 | 类型 | 默认值 |
|:---|:---|:---:|:---:|
| type| 类型，可选值为`primary` `light` `plain` `grey` `link` | string | `primary` |
| size | 尺寸，可选值为`large` `medium` `subMedium` `normal` `small` `mini` | string | `normal` |
| disabled | 是否禁用按钮 | boolean | `false` |
| square | 是否为方形按钮 | boolean | `false` |
| round | 是否为圆形按钮 | boolean | `false` |
| loading | 是否显示为加载状态 | boolean | `false`
| loading-text | 加载状态提示文字 | string | - |
| icon | 图标链接 | string | - |
| icon-text | 图标提示文字 | string | - |
| customClass | 自定义类名 | string | - |

</card>

<card>

### Events

| 事件名 | 说明 | 回调参数 |
|:---|:---|:---|
| click | 点击按钮，按钮为禁用或加载状态下也会触发 | _event: Event_ |

</card>

<demo/>
