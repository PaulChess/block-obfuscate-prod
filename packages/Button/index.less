@import '../cssVars.less';
@import './var.less';

// clear button's default font-size
button {
  font-size: inherit;
}

.hxmui-button {
  position: relative;
  display: inline-flex;
}

.hxm-button {
  position: relative;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  text-align: center;
  appearance: none;
  background-color: transparent;
  font-weight: var(--button-weight);
  border-radius: var(--button-border-radius);

  &::before {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border: inherit;
    border-radius: inherit;
    background-color: black;
    opacity: 0;
    content: ' ';
  }

  &:active:enabled::before {
    opacity: 0.1;
  }

  // button-type
  &--primary {
    background-color: var(--button-primary-background-color);
    color: var(--button-primary-color);

    [theme-mode="black"] & {
      background-color: #fd4332;
      color: #fff;
    }

    &:active::before {
      background-color: #000;
    }

    [theme-mode="black"] &:active::before {
      background-color: #fff;
    }
  }

  &--light {
    background-color: var(--button-light-background-color);
    color: var(--button-light-color);

    [theme-mode="black"] & {
      background-color: rgba(253, 67, 50, 0.1);
      color: #fd4332;
    }

    &:active::before {
      background-color: #000;
    }

    [theme-mode="black"] &:active::before {
      background-color: #fff;
    }
  }

  &--plain {
    background-color: transparent;
    border: var(--button-border-width) solid var(--button-plain-border-color);
    color: var(--button-plain-color);

    [theme-mode="black"] & {
      border: 0.01em solid #fd4332;
      color: #fd4332;
    }

    &:active::before {
      background-color: var(--button-plain-border-color);
    }

    [theme-mode="black"] &:active::before {
      background-color: #fd4332;
    }
  }

  &--grey {
    background-color: transparent;
    border: var(--button-border-width) solid var(--button-grey-border-color);
    color: var(--button-grey-color);

    [theme-mode="black"] & {
      border: 0.01em solid #8e8e8e;
      color: #a9a9a9;
    }

    &:active::before,
    [theme-mode="black"] &:active::before {
      background-color: var(--button-grey-color);
    }
  }

  &--link {
    background-color: transparent;

    &:active::before,
    [theme-mode="black"] &:active::before {
      background-color: transparent;
    }

    color: var(--hxm-blue);
  }

  // button-size
  &--large {
    width: 100%;
    height: var(--button-large-height);
    line-height: var(--button-large-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-large-font-size);
    }
  }

  &--medium {
    width: 100%;
    height: var(--button-medium-height);
    line-height: var(--button-medium-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-medium-font-size);
    }
  }

  &--subMedium {
    box-sizing: border-box;
    padding: 0 var(--button-submedium-padding);
    height: var(--button-submedium-height);
    line-height: var(--button-submedium-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-submedium-font-size);
    }
  }

  &--normal {
    box-sizing: border-box;
    padding: 0 var(--button-normal-padding);
    height: var(--button-normal-height);
    line-height: var(--button-normal-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-normal-font-size);
    }
  }

  &--small {
    box-sizing: border-box;
    padding: 0 var(--button-small-padding);
    height: var(--button-small-height);
    line-height: var(--button-small-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-small-font-size);
    }
  }

  &--mini {
    box-sizing: border-box;
    padding: 0 var(--button-mini-padding);
    height: var(--button-mini-height);
    line-height: var(--button-mini-height);

    .text,
    .loading-text,
    .icon-text {
      font-size: var(--button-mini-font-size);
    }
  }

  // shape
  &--square {
    border-radius: 0;
  }

  &--round {
    border-radius: var(--button-round-border-radius);
  }

  // disabled
  &--disabled {
    pointer-events: none;
    cursor: not-allowed;
  }
  // ���ڲ�ͬ�İ�ť�����ڽ�ֹ״̬����ʽҲ����ͬ�����Ҳ��Ҫ�ֱ�����
  &--disabled&--primary {
    background-color: var(--button-primary-disabled-background-color);
    color: var(--button-primary-disabled-text-color);
  }

  [theme-mode="black"] &--disabled&--primary {
    background-color: rgba(253, 67, 50, 0.3);
    color: rgba(255, 255, 255, 0.1);
  }

  &--disabled&--plain {
    border: var(--button-border-width) solid var(--button-plain-disabled-border-color);
    color: var(--button-plain-disabled-text-color);
  }

  [theme-mode="black"] &--disabled&--plain {
    border: 0.01em solid rgba(253, 67, 50, 0.3);
    color: rgba(253, 67, 50, 0.3);
  }

  &--disabled&--grey {
    border: var(--button-border-width) solid var(--button-grey-disabled-border-color);
    color: var(--button-grey-disabled-text-color);
  }

  [theme-mode="black"] &--disabled&--grey {
    border: 0.01em solid rgba(169, 169, 169, 0.3);
    color: rgba(169, 169, 169, 0.3);
  }

  &--disabled&--link {
    color: var(--button-link-disabled-text-color);
  }

  [theme-mode="black"] &--disabled&--link {
    color: rgba(61, 118, 184, 0.3);
  }

  // loading
  &--loading {
    pointer-events: none;
  }

  &__content {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
    pointer-events: auto;

    .loading-block,
    .icon-block {
      display: flex;
      justify-content: center;
      align-items: center;
    }

    .icon-block {
      .icon {
        margin-right: 0.04em;
      }
    }
  }
}
