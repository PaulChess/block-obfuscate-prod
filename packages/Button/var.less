:root {
  --button-primary-color: #fff;
  --button-primary-background-color: var(--hxm-primary-1);
  --button-primary-disabled-background-color: rgba(233, 48, 48, 0.3);
  --button-primary-disabled-text-color: rgba(255, 255, 255, 0.6);
  --button-light-color: var(--hxm-primary-1);
  --button-light-background-color: rgba(233, 48, 48, 0.1);
  --button-plain-color: var(--hxm-primary-1);
  --button-plain-border-color: var(--hxm-primary-1);
  --button-plain-disabled-border-color: rgba(233, 48, 48, 0.3);
  --button-plain-disabled-text-color: rgba(233, 48, 48, 0.3);
  --button-grey-color: var(--hxm-text-2);
  --button-grey-border-color: var(--hxm-text-3);
  --button-grey-disabled-border-color: rgba(102, 102, 102, 0.3);
  --button-grey-disabled-text-color: rgba(102, 102, 102, 0.3);
  --button-link-disabled-text-color: rgba(70, 145, 238, 0.3);
  --button-weight: var(--font-weight-normal);
  --button-mini-height: 0.28em;
  --button-small-height: 0.32em;
  --button-normal-height: 0.36em;
  --button-submedium-height: 0.44em;
  --button-medium-height: 0.44em;
  --button-large-height: 0.44em;
  --button-mini-padding: 0.1em;
  --button-small-padding: 0.22em;
  --button-normal-padding: 0.24em;
  --button-submedium-padding: 0.9em;
  --button-mini-font-size: var(--font-size-mini);
  --button-small-font-size: var(--font-size-small);
  --button-normal-font-size: var(--font-size-normal);
  --button-medium-font-size: var(--font-size-large);
  --button-submedium-font-size: var(--font-size-large);
  --button-large-font-size: var(--font-size-large);
  --button-border-width: 0.01em;
  --button-border-radius: 0.04em;
  --button-round-border-radius: 1em;
}

[theme-mode="black"] {
  --button-primary-color: #fff;
  --button-primary-background-color: var(--hxm-primary-1);
  --button-primary-disabled-background-color: rgba(253, 67, 50, 0.3);
  --button-primary-disabled-text-color: rgba(255, 255, 255, 0.1);
  --button-light-color: var(--hxm-primary-1);
  --button-light-background-color: rgba(253, 67, 50, 0.1);
  --button-plain-color: var(--hxm-primary-1);
  --button-plain-border-color: var(--hxm-primary-1);
  --button-plain-disabled-border-color: rgba(253, 67, 50, 0.3);
  --button-plain-disabled-text-color: rgba(253, 67, 50, 0.3);
  --button-grey-color: var(--hxm-text-2);
  --button-grey-border-color: var(--hxm-text-3);
  --button-grey-disabled-border-color: rgba(169, 169, 169, 0.3);
  --button-grey-disabled-text-color: rgba(169, 169, 169, 0.3);
  --button-link-disabled-text-color: rgba(61, 118, 184, 0.3);
}
