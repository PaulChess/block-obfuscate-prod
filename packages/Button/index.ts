import Button from './Button.vue';
import install from '../utils/install';

export default install(Button);
