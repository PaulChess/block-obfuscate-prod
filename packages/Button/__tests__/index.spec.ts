/*
 * @Author: liushuai
 * @Date: 2021-07-14 15:51:03
 * @LastEditTime: 2021-07-15 17:07:02
 */
import { mount } from '@vue/test-utils';
import Button from '../Button.vue';

const m = (props: object) => mount(Button, { propsData: { ...props } });
const matchSnap = (fn: (props: object) => any, props: object) => expect(fn(props).html()).toMatchSnapshot();

describe('Button', () => {
  /** event */
  it('should emit click event', async () => {
    const wrapper = mount(Button);

    wrapper.find('button').trigger('click');

    expect(wrapper.emitted().click).toBeTruthy();
    expect(wrapper.emitted().click?.length).toEqual(1);
  });

  it('should not emit click event when disabled', async () => {
    const wrapper = mount(Button, {
      propsData: {
        disabled: true,
      },
    });

    wrapper.find('button').trigger('click');

    expect(wrapper.emitted().click).toBeFalsy();
  });

  it('should not emit click event when loading', () => {
    const wrapper = mount(Button, {
      propsData: {
        loading: true,
      },
    });

    wrapper.find('button').trigger('click');

    expect(wrapper.emitted().click).toBeFalsy();
  });

  /** props */
  it('props loading', () => {
    matchSnap(m, {
      loading: true,
      loadingText: '加载中...',
    });
  });

  it('props custom class', () => {
    matchSnap(m, { customClass: 'attention' });
  });

  it('props type', () => {
    matchSnap(m, { type: 'primary' });
    matchSnap(m, { type: 'plain' });
    matchSnap(m, { type: 'grey' });
    matchSnap(m, { type: 'link' });
  });

  it('props size', () => {
    matchSnap(m, { size: 'large' });
    matchSnap(m, { size: 'medium' });
    matchSnap(m, { size: 'subMedium' });
    matchSnap(m, { size: 'normal' });
    matchSnap(m, { size: 'small' });
    matchSnap(m, { size: 'mini' });
  });

  it('props icon', () => {
    matchSnap(m, { type: 'plain', icon: 'https://testm.10jqka.com.cn/cd/mbweb-vaserviece-strategy-api-container/newjuece_v3/img/top1.89c5c19.png', iconText: '榜1' });
  });

  it('props square', () => {
    matchSnap(m, { square: 'true' });
    matchSnap(m, { round: 'true' });
  });
});
