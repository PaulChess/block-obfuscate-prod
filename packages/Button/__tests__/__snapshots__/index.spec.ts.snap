// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button props custom class 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--normal attention">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props icon 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--plain hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <div class="icon-block"><img src="https://testm.10jqka.com.cn/cd/mbweb-vaserviece-strategy-api-container/newjuece_v3/img/top1.89c5c19.png" class="icon" style="width: 0.16em;"> <span class="icon-text">榜1</span></div>
    </div>
  </button></div>
`;

exports[`Button props loading 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button disabled="disabled" class="hxm-button hxm-button--primary hxm-button--normal hxm-button--loading">
    <div class="hxm-button__content">
      <!---->
      <div class="loading-block">
        <div class="hxmui-loading hxmui-fontsize-adapter">
          <div class="hxm-loading">
            <div class="hxm-loading__spinner hxm-loading__spinner--circle" style="width: 0.16em; height: 0.16em; color: rgb(255, 255, 255);"><svg viewBox="25 25 50 50" class="hxm-loading__circle">
                <circle cx="50" cy="50" r="20" fill="none"></circle>
              </svg></div> <span class="hxm-loading__text"><span class="hxm-loading__text--inner"></span></span>
          </div>
        </div> <span class="loading-text">加载中...</span>
      </div>
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter" style="width: 100%;"><button class="hxm-button hxm-button--primary hxm-button--large">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 2`] = `
<div class="hxmui-button hxmui-fontsize-adapter" style="width: 50%;"><button class="hxm-button hxm-button--primary hxm-button--medium">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 3`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--subMedium">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 4`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 5`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--small">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props size 6`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--mini">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props square 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--normal hxm-button--square">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props square 2`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--normal hxm-button--round">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props type 1`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--primary hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props type 2`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--plain hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props type 3`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--grey hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;

exports[`Button props type 4`] = `
<div class="hxmui-button hxmui-fontsize-adapter"><button class="hxm-button hxm-button--link hxm-button--normal">
    <div class="hxm-button__content"><span class="text"></span>
      <!---->
      <!---->
    </div>
  </button></div>
`;
